variables.tf

```
@@ -1,6 +1,11 @@
 locals {
   windows_license_types = ["None", "Windows_Client", "Windows_Server"]
-  linux_license_types   = ["RHEL_BYOS", "RHEL_BASE", "RHEL_EUS", "RHEL_SAPAPPS", "RHEL_SAPHA", "RHEL_BASESAPAPPS", "RHEL_BASESAPHA", "SLES_BYOS", "SLES_SAP", "SLES_HPC"]
+  linux_license_types = [
+    "RHEL_BYOS", "RHEL_BASE", "RHEL_EUS", "RHEL_SAPAPPS", "RHEL_SAPHA", "RHEL_BASESAPAPPS", "RHEL_BASESAPHA",
+    "SLES_BYOS", "SLES_SAP", "SLES_HPC"
+  ]
+  identity_type = (var.entra_id_login.enabled ? (try(var.identity.type, "") == "UserAssigned" ?
+  replace(var.identity.type, "UserAssigned", "SystemAssigned, UserAssigned") : "SystemAssigned") : "")
```

This logic is quite complex and difficult to read. I recommend reformatting it for better clarity:

```suggestion
  identity_type = (
    var.entra_id_login.enabled ? (
      try(var.identity.type, "") == "UserAssigned" ?
      replace(var.identity.type, "UserAssigned", "SystemAssigned, UserAssigned") :
      "SystemAssigned"
    ) :
  "")
```

Additionally, the logic itself feels a bit overwhelming. What do you think about the following alternative?

```suggestion
  # Add "SystemAssigned" to the identity type if EntraID is enabled
  # Otherwise fallback to var.identity.type
  identity_type = (
    !var.entra_id_login.enabled ? var.identity.type :
    join(", ", distinct(concat(["SystemAssigned"], split(", ", var.identity.type))))
  )
```

With this alternative, one can always use `local.identity_type` because it considers to `var.identity.type` (as mentioned in the comment above)

---

I am not sure if this works, as can  get "UserAssigned, SystemAssigned", which is the worng order, according to tge docu https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#type-1

---

Actually, you don't – see:

```console
$ terraform console -var=identity='{type="UserAssigned"}'
> join(", ", distinct(concat(["SystemAssigned"], split(", ", var.identity.type))))
"SystemAssigned, UserAssigned"
```

```console
$ terraform console -var=identity='{type="SystemAssigned"}'
> join(", ", distinct(concat(["SystemAssigned"], split(", ", var.identity.type))))
"SystemAssigned"
```

```console
$ terraform console -var=identity='{type="SystemAssigned, UserAssigned"}'
> join(", ", distinct(concat(["SystemAssigned"], split(", ", var.identity.type))))
"SystemAssigned, UserAssigned"
```